src/test/java/bio/terra/common/UriUtilsTest.java

```diff
@@ -0,0 +1,74 @@
+package bio.terra.common;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import bio.terra.common.category.Unit;
+import bio.terra.service.filedata.exception.InvalidUriException;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(Unit.class)
+public class UriUtilsTest {
+
+  @Test
+  public void testGetQueryParameterValue() {
+    assertThat(
+        "query parameter is returned",
+        UriUtils.getValueFromQueryParameter("http://foo/bar?q=123", "q"),
+        equalTo("123"));
+    assertThat(
+        "query parameter is not found",
+        UriUtils.getValueFromQueryParameter("http://foo/bar", "q"),
+        is(nullValue()));
+    assertThat(
+        "multivalues are aggregated",
+        UriUtils.getValueFromQueryParameter("http://foo/bar?q=123&q=456", "q"),
+        equalTo("123,456"));
+  }
+
+  @Test
+  public void testInvalidUris() {
+    String badUri = "not a good uri";
+
+    TestUtils.assertError(
+        InvalidUriException.class,
+        "Invalid URI",
+        () -> UriUtils.getValueFromQueryParameter(badUri, "q"));
+
+    TestUtils.assertError(
```

Comment:
I think this is a duplicate of the assertion right above it

Reply:
doh!  It sure is